### PR TITLE
Autocomplete and RawAutocomplete initialValue parameter

### LINKED ISSUE #78179: [Autocomplete] Expose Autocomplete.textEditingController parameter like RawAutocomplete
As titled. This allows setting the initial value for the `Autocomplete` widget

### PR DESCRIPTION
**Pull request by shihaohong**
Provides `initialValue` as a parameter for `Autocomplete` and `RawAutocomplete` widgets. This allows users to set the initial value of their text fields to the String provided. For `RawAutocomplete`, both `initialValue` and `textEditingController` cannot be simultaneously defined since the `textEditingController` should have its text value defined when passed into `RawAutocomplete` rather than depending on the `initialValue`.

I thought about just wiring `textEditingController` from `Autocomplete` to `RawAutocomplete`, but the main purpose of that API was to support the split UI feature and that isn't a concern for the `Autocomplete` widget, so having the requirement of having pass in a FocusNode along with a text editing controller doesn't make sense (a focus node is required if a text editing controller is passed in).

Fixes https://github.com/flutter/flutter/issues/78179.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by shihaohong** [comment]
@LongCatIsLooong could you take a look at this since @justinmc is on leave?

File: packages/flutter/lib/src/widgets/autocomplete.dart [hunk 1]
```diff
@@ -811,7 +821,7 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
   @override
   void initState() {
     super.initState();
-    _textEditingController = widget.textEditingController ?? TextEditingController();
+    _textEditingController = widget.textEditingController ?? TextEditingController(text: widget.initialValue);
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/autocomplete.dart:824]: Maybe set the selection to the end of the string, or expose `TextEditingValue` instead? IIRC an invalid selection can spell trouble if the text input is auto-focused (also I'm planning to stop painting the caret in `RenderEditable` if an invalid selection is specified: https://github.com/flutter/flutter/pull/79607).

File: packages/flutter/lib/src/material/autocomplete.dart [hunk 2]
```diff
@@ -192,6 +193,11 @@ class Autocomplete<T extends Object> extends StatelessWidget {
   /// default.
   final AutocompleteOptionsViewBuilder<T>? optionsViewBuilder;
 
+  /// The initial value to use for the text field.
+  ///
+  /// This parameter is ignored if [textEditingController] is defined.
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/autocomplete.dart:198]: nit: maybe use a macro for this?

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/autocomplete.dart:198]: It looks like setting the initial value does not notify `textEditingController`'s listeners, so the dropdown menu won't show (maybe worth documenting?)

File: packages/flutter/lib/src/widgets/autocomplete.dart (hunk 1 above)
Comment (shihaohong) [reply packages/flutter/lib/src/widgets/autocomplete.dart:824]: I opted to expose TextEditingValue

**Review by LongCatIsLooong — APPROVED** [review]

**Pull request merged**